src/Iyzipay/Model/Customer.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+namespace Iyzipay\Model;
+
+use Iyzipay\BaseModel;
+use Iyzipay\JsonBuilder;
+use Iyzipay\RequestStringBuilder;
+
+class Customer extends BaseModel
+{
+    private $name;
+    private $surname;
+    private $identityNumber;
+    private $email;
+    private $gsmNumber;
+    private $shippingAddressContactName;
+    private $shippingAddressCity;
+    private $shippingAddressCountry;
+    private $shippingAddressAddress;
+    private $shippingAddressZipCode;
+    private $billingAddressContactName;
+    private $billingAddressCity;
+    private $billingAddressCountry;
+    private $billingAddressAddress;
+    private $billingAddressZipCode;
+
+
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    public function setName($name)
+    {
+        $this->name = $name;
+    }
+
+    public function getSurname()
+    {
+        return $this->surname;
+    }
+
+    public function setSurname($surname)
+    {
+        $this->surname = $surname;
+    }
+
+    public function getIdentityNumber()
+    {
+        return $this->identityNumber;
+    }
+
+    public function setIdentityNumber($identityNumber)
+    {
+        $this->identityNumber = $identityNumber;
+    }
+
+    public function getEmail()
+    {
+        return $this->email;
+    }
+
+    public function setEmail($email)
+    {
+        $this->email = $email;
+    }
+
+    public function getGsmNumber()
+    {
+        return $this->gsmNumber;
+    }
+
+    public function setGsmNumber($gsmNumber)
+    {
+        $this->gsmNumber = $gsmNumber;
+    }
+
+    public function getShippingAddressContactName(){
```

Comment:
Çoktan merge edildi farkındayım ama shipping ile ilgili fieldları \Iyzipay\Model\Address modeli üzerinden assign etsek daha sağlıklı olmaz mı?

Reply:
Örneklerde hep aşağıdaki gibi bir kullanım var.
$request->setBillingAddress($billingAddress);
$request->setShippingAddress($shippingAddress);

Reply:
Bir laravel paketi geliştirmekteyim ve user modelini iyzico customer modeline çevirirken aşağıdaki gibi hoş olmayan şekilde kullanımlar oluyor.

setBillingAddressAddress
setShippingAddressAddress 


```php
    public function toIyzicoCustomer(): Customer
    {
        $buyer = new Customer();
        $buyer->setName($this->name);
        $buyer->setSurname($this->surname);
        $buyer->setEmail($this->email);
        $buyer->setGsmNumber($this->phone_number);
        $buyer->setIdentityNumber($this->identity_number);

        $buyer->setShippingAddressContactName($this->name. ' ' . $this->surname);
        $buyer->setShippingAddressCity($this->city);
        $buyer->setShippingAddressCountry($this->country);
        $buyer->setShippingAddressAddress($this->address);
        $buyer->setShippingAddressZipCode($this->zipcode);

        $buyer->setBillingAddressContactName($this->name. ' ' . $this->surname);
        $buyer->setBillingAddressCity($this->city);
        $buyer->setBillingAddressCountry($this->country);
        $buyer->setBillingAddressAddress($this->address);
        $buyer->setBillingAddressZipCode($this->zipcode);

        return $buyer;
    }

```

Reply:
Selamlar Talha, 
Teşekkürler yorumun için.  Customer sınıfını address sınıfından bağımsız olarak oluşturudum. Bu yüzden request içine shipping ve billing adres yerine customer set ediliyor ve customer içinde adres bilgisi gönderiliyor. Belirttiğin değerlerin karışıklığına katılıyorum. Bir sonraki pr'da burada düzeltme yapacağım. Eğer katkıda bulunmak istersen PR açabilirsin. Review edip mergeleyebiliriz.